File: DS/Stacks/Python/postfix_evaluation.py
def postfix(str1, n):
    stack = []

    for i in range(n):
        if str1[i].isnumeric():
            stack.append(int(str1[i]))

        else:
            if str1[i] == '+':
                op2 = stack.pop()
                op1 = stack.pop()
                stack.append(op1 + op2)

            elif str1[i] == '-':
                op2 = stack.pop()
                op1 = stack.pop()
                stack.append(op1 - op2)

            elif str1[i] == '*':
                op2 = stack.pop()
                op1 = stack.pop()
                stack.append(op1 * op2)

            elif str1[i] == '/':
                op2 = stack.pop()
                op1 = stack.pop()
                stack.append(op1 / op2)

    return stack.pop()


s = "23*54*+9-"
print(postfix(s, len(s)))

# output: 17

File: DS/Stacks/cpp/postfix_evaluation.cpp
#include <bits/stdc++.h>
using namespace std;

int postfixEvaluation(string s){
    stack<int> st;
    int a,b;

    for(int i=0; i<s.length();i++){
        if(isdigit(s[i])){
            st.push(s[i]-'0');
        }

        else{
            switch (s[i])
            {
            case '*':
                a = st.top();
                st.pop();
                b = st.top();
                st.pop();
                st.push(b*a);
                break;
            
            case '/':
                a = st.top();
                st.pop();
                b = st.top();
                st.pop();
                st.push(b/a);
                break;
            
            case '+':
                a = st.top();
                st.pop();
                b = st.top();
                st.pop();
                st.push(b+a);
                break;

            case '-':
                a = st.top();
                st.pop();
                b = st.top();
                st.pop();
                st.push(b-a);
                break;

            default:
                break;
            }
        }
    }
    
    return st.top();
}

int main(){
    string s = "23*54*+9-";
    cout<<postfixEvaluation(s);
}

// output: 17